plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace edsl;
+
+namespace {
+
+// Compute pad_before and the size of output.
+// This function is based on (the functon compute_padding_and_output_size() in plaidml/op/lib/ops.cc).
+// The type of the return variable and parameters are changed from TensorDim to size_t, because TensorDim can't be
+// converted to int.
+std::pair<size_t, size_t> compute_padding_before_and_output_size(size_t input_size, size_t off_size, size_t filter_size,
+                                                                 size_t stride, plaidml::op::AutoPadMode autopad_mode,
+                                                                 size_t pad_before, size_t pad_end, size_t dilation) {
+  size_t I_eff = input_size;
+  size_t F_eff = (dilation * (filter_size - 1)) + 1;
+  if (autopad_mode == plaidml::op::AutoPadMode::EXPLICIT) {
+    size_t output_size = ((I_eff + pad_before + pad_end - F_eff + stride) / stride);
+    std::pair<size_t, size_t> result(pad_before, output_size);
+    return result;
+  }
+  if (autopad_mode == plaidml::op::AutoPadMode::VALID) {
+    size_t output_size = ((I_eff - F_eff + stride) / stride);
+    std::pair<size_t, size_t> result(0, output_size);
+    return result;
+  }
+  if (autopad_mode == plaidml::op::AutoPadMode::SAME_LOWER || autopad_mode == plaidml::op::AutoPadMode::SAME_UPPER) {
+    size_t lower_term = (autopad_mode == plaidml::op::AutoPadMode::SAME_LOWER) ? 1 : 0;
+    size_t max = 0;
+    if (((off_size - 1) * stride + F_eff - input_size) > 0) {
+      max = (off_size - 1) * stride + F_eff - input_size;
+    }
+    pad_before = (max + lower_term) / 2;
+    size_t output_size = ((I_eff + stride - 1) / stride);
+    std::pair<size_t, size_t> result(pad_before, output_size);
+    return result;
+  }
+  THROW_IE_EXCEPTION << "Unexpected autopadding mode.";
+}
+
+std::vector<int> cast_vector(std::vector<int64_t> vec) {
```

Comment:
Is `cast_vector` needed? It looks like these vectors are just being used as parameters in `compute_deformable_convolution ` -- can we not just change the type in that function to vectors of int64_t instead of vectors of ints?

Reply:
I use cast_vector because that the function op::broadcast just supports vector<int> as parameter, if i don't use cast_vector, there will be too many static_cast<int> function when i use the function op::broadcast.

Reply:
Makes sense, seems ok. If you think it would simplify this code in other ways, I don't see any problems with changing `broadcast` to use `int64_t` instead of `int`. But your current approach is also fine.

Reply:
Done, we have removed the `cast_vector` function and changed the `op::broadcast` to use `int64_t` instead of `int` and fixed all related tests.